Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -41,10 +41,7 @@ public extension Sequence {
     /// - Parameter condition: condition to evaluate each element against.
     /// - Returns: the last element in the array matching the specified condition. (optional)
     func last(where condition: (Element) throws -> Bool) rethrows -> Element? {
-        for element in reversed() {
-            if try condition(element) { return element }
-        }
-        return nil
+        return try reversed().first(where: condition)
```

Comment:
Seems like there's no much we could do here to avoid the copy of `reversed()` the only other implementation I could think of would be iterate forward over all items ... so we would basically trade avoid copy for a slower version ...

Reply:
Some sequences have a lazy-esque version of `reversed`. I'd like to think that Swift will call them, instead of creating the copy.

Reply:
Yeah, but I think the only other version we have of `reversed` that is a lazy view is down to `BidirectionalCollection` ... in the case of this function the selected overload will always be this copy one of `Sequence`... but for this case is fine since `BidirectionalCollection` has it's own version of `last(where:)` which is index based and can iterate backwards.

Reply:
Makes sense that it can only from bidirectional it can be lazified. I guess better to improve performance than write shorter code.